network/client/Client.cpp

```diff
@@ -0,0 +1,175 @@
+/*
+** EPITECH PROJECT, 2024
+** R-Type-Reborn
+** File description:
+** No file there , just an epitech header example .
+** You can even have multiple lines if you want !
+*/
+
+#include "Client.hpp"
+#include <iostream>
+#include <utility>
+#include <asio/ip/udp.hpp>
+#include <thread>
+
+//-------------------------------------Initiator------------------------------------------
+Network::Client::Client(std::string host, const unsigned short udp_port, unsigned short tcp_port)
+        : _host(std::move(host)), _UDP_PORT(udp_port), _TCP_PORT(tcp_port), _udp_socket(_io_context), _tcp_socket(_io_context), _is_alive(true)
+{
+    _id = -1;
+    _send_timer = std::make_shared<asio::steady_timer>(_io_context, std::chrono::milliseconds(15));
+    _recv_buffer.resize(2048);
+}
+
+
+
+void Network::Client::connect()
+{
+
+    // TCP connection init
+    asio::ip::tcp::resolver resolver(_io_context);
+    asio::ip::tcp::resolver::results_type const endpoints = resolver.resolve(_host, std::to_string(_TCP_PORT));
+    asio::error_code error;
+    _tcp_endpoint = asio::connect(_tcp_socket, endpoints, error);
+    if (error) {
+        std::cerr << "Error: " << error.message() << std::endl;
+        return;
```

Comment:
All your error should use throw (except if it's not a critical error)
Create a NetworkException that inherit from IError which is present in includes/stellarforge/common